### ISSUE TITLE
vulnerability in ractive.link

### ISSUE THREAD

**Issue by Jeff17Robbins** [ractivejs/ractive#3107]
## Description: 
Using the `{ractive: sourceRactive}` option, and passing `undefined` (by mistake) as `sourceRactive` the second time `ractive.link()` is called, crashes with

```
Uncaught RangeError: Maximum call stack size exceeded
```

## Reproduction:

https://jsfiddle.net/rxj6fcn4/

```
var source = Ractive({
  data: {
    loggedIn: false
  }
});

var destination = Ractive();

destination.link('loggedIn', 'loggedIn', {ractive: source});

// this (mistake) line causes
// Uncaught RangeError: Maximum call stack size exceeded
destination.link('loggedIn', 'loggedIn', {ractive: undefined});
```

Making the source and destination keypaths different prevents the crash: https://jsfiddle.net/rxj6fcn4/2/

```
var source = Ractive({
  data: {
    loggedIn: false
  }
});

var destination = Ractive();

destination.link('loggedIn', 'xloggedIn', {ractive: source});

destination.link('loggedIn', 'xloggedIn', {ractive: undefined});
```

**Comment by evs-chris** [comment]
This is now fixed in edge and will go out with 0.9.6 when it's cut. Thanks for the report!

**Comment by Jeff17Robbins** [comment]
Thanks!
